x/protocolpool/keeper/genesis.go

```diff
@@ -49,8 +51,21 @@ func (k Keeper) InitGenesis(ctx context.Context, data *types.GenesisState) error
 		}
 	}
 
-	if err := k.ToDistribute.Set(ctx, data.ToDistribute); err != nil {
-		return fmt.Errorf("failed to set to distribute: %w", err)
+	if err := k.LastBalance.Set(ctx, data.LastBalance); err != nil {
+		return fmt.Errorf("failed to set last balance: %w", err)
+	}
+
+	totalToBeDistributed := math.ZeroInt()
+	for _, distribution := range data.Distributions {
+		totalToBeDistributed = totalToBeDistributed.Add(distribution.Amount)
+		if err := k.Distributions.Set(ctx, *distribution.Time, distribution.Amount); err != nil {
```

Comment:
why do we need to keep track of when the distribution is done?

Reply:
Because a continuous fund could expire in between, so we keep track of distributions received at t, t+1, t+n..., so when we distribute to the continuous funds, a fund expiring at t+2 will receive the distributions made until that point, and not more

Reply:
https://github.com/cosmos/cosmos-sdk/pull/20790/files#diff-dcf594a333e20d0c216d697c5b17b3a407ca56b2f8acd2f6f1e4a13904f5a513R207-R213